src/server.py

```diff
@@ -464,37 +420,97 @@ def run_seta_details_query():
 
+        # we query all jobs rather than jobs filter by the requested priority in here,
+        # Because we need to set the job returning strategy depend on different job priority.
+        query = session.query(JobPriorities.platform,
+                              JobPriorities.buildtype,
+                              JobPriorities.testtype,
+                              JobPriorities.priority,
+                              JobPriorities.timeout
+                              ).all()
+
         # We should return full job list as a fallback, if it's a request from
         # taskcluster and without head_rev or pushlog_id in there
-        if head_rev or pushlog_id:
-            priority = valve(head_rev, pushlog_id, branch, priority)
+        branch_info = session.query(TaskRequests.counter, TaskRequests.datetime).filter(
+            TaskRequests.branch == branch).all()
+
+        time_of_now = datetime.datetime.now()
+        time_of_request = time.strftime("%Y-%m-%d %H:%M:%S")
+
+        # If we got nothing related with that branch, we should create it.
+        if len(branch_info) == 0:
+            # time_of_lastreset is not a good name anyway :(
+            # And we treat all branches' reset_delta is 90 seconds, we should find a
+            # better delta for them in the further.
+            branch_data = TaskRequests(str(branch), 1, str(time_of_request), 5400)
+            try:
+                session.add(branch_data)
+                session.commit()
+            except Exception:
+                session.rollback()
+
+            finally:
+                session.close()
+            counter, string_time, reset_delta = 1, time_of_request, 5400
+
+        # We should update it if that branch had already been stored.
         else:
-            priority = 0
+            counter, string_time, reset_delta = branch_info[0]
+            conn = engine.connect()
+            statement = update(TaskRequests).where(
+                TaskRequests.branch == branch).value(
+                counter == counter + 1)
+            conn.execute(statement)
+
+        last_update_time = datetime.datetime.strptime(string_time, "%Y-%m-%d %H:%M:%S")
+        delta = (time_of_now - last_update_time).total_seconds()
+
+        # we should update the time recorder if the elapse time had
+        # reach the time limit of that branch.
+        if delta >= reset_delta:
+            conn = engine.connect()
+            statement = update(TaskRequests).where(
+                TaskRequests.branch == branch).value(
+                datetime == datetime)
+            conn.execute(statement)
+
+        for d in query:
+            # we only return that job if it hasn't reach the timeout limit.
+            if delta < d[4]:
+                # Due to the priority of all high value jobs is 1, and we
+                # need to return all jobs for every 5 pushes(for now).
+                if counter + 1 % d[3] != 0:
```

Comment:
if we have a variable counter and increment it originally- we can add that counter to the database and just use it here  without (counter +1).